staging/kos/pkg/controllers/ipam/ipam.go

```diff
@@ -829,12 +851,33 @@ func (ctlr *IPAMController) updateNAStatus(ns, name string, att *netv1a1.Network
 	tAfter := time.Now()
 	ctlr.attachmentUpdateHistograms.With(prometheus.Labels{"statusErr": FmtErrBool(len(statusErrs) > 0), "err": SummarizeErr(err)}).Observe(tAfter.Sub(tBefore).Seconds())
 	if err == nil {
-		deltaS := att3.Writes.GetServerWriteTime(netv1a1.NASectionAddr).Sub(att.Writes.GetServerWriteTimeUnwrapped(netv1a1.NASectionSpec)).Seconds()
-		ctlr.attachmentCreateToAddressHistograms.With(prometheus.Labels{contentionLabel: strconv.FormatBool(nadat.addressContention)}).Observe(deltaS)
+		deltaSecs := "unknown"
+		if (subnetCreatTime != time.Time{}) {
```

Comment:
Aha, I thought there were still problems with this histogram.  It was being updated in some inappropriate circumstances.  I think the right test is whether `len(statusErrs) == 0`, or equivalently, `ipForStatus != nil`.

Reply:
Done in 52413b4 , but I was planning (and still do) a PR that tracks the "update without address" case as well via a label.